letsencrypt-win-simple/Plugin/SanByDomainPlugin.cs

```diff
@@ -0,0 +1,288 @@
+                                target.SiteId = item.SiteId;
+                                siteList.Add(target);
+                            }
+
+                            Target totalTarget = CreateTarget(siteList);
+                            ProcessTotaltarget(totalTarget, siteList);
+                        }
+                    }
+                    else
+                    {
+                        Console.WriteLine("Invalid Id...");
+                    }
+                }
+                catch (Exception ex)
+                {
+                    Console.WriteLine("An error occured!");
+                    Console.ForegroundColor = ConsoleColor.Red;
+                    Console.WriteLine(ex.Message);
+                    Console.ResetColor();
+                }
+            }
+        }
+
+        public override void Auto(Target target)
+        {
+            Console.WriteLine("Auto isn't supported for SanByDomain Plugin");
+        }
+
+        public override void Renew(Target target)
+        {
+            Console.WriteLine("Renew isn't supported for SanByDomain Plugin");
+        }
+
+        //Mostly copied from IISSiteServerPlugin
+        private Target CreateTarget(List<Target> sites)
+        {
+            Target totalTarget = new Target();
+            totalTarget.PluginName = Name;
+            totalTarget.SiteId = 0;
+            totalTarget.WebRootPath = "";
+
+            foreach (var site in sites)
+            {
+                var auth = Program.Authorize(site);
+                if (auth.Status != "valid")
+                {
+                    Console.WriteLine("All hosts under all sites need to pass authorization before you can continue");
+                    Log.Error("All hosts under all sites need to pass authorization before you can continue.");
+                    //If Environment.Exit(1) is executed here the error message will not be shown since the window closes!!!
+                    //If not - the app continues and it will try to get certificates without auth...
+                    //Environment.Exit(1);
+                }
+                else
+                {
+                    if (totalTarget.Host == null)
+                    {
+                        totalTarget.Host = site.Host;
+                    }
+
+                    if (totalTarget.AlternativeNames == null)
+                    {
+                        List<string> hostList = new List<string>();
+                        hostList.Add(site.Host);
+                        totalTarget.AlternativeNames = hostList;
+                    }
+                    else
+                    {
+                        totalTarget.AlternativeNames.Add(site.Host);
+                    }
+                }
+            }
+            return totalTarget;
+        }
+
+        //Copied from IISSiteServerPlugin
+        private static void ProcessTotaltarget(Target totalTarget, List<Target> runSites)
```

Comment:
Since you are doing the exact same thing, and your plugin really is doing the same thing as the IIS Site Server plugin, but doing it with domains instead of multiple sites, you should just use the same function. Otherwise there is duplicated code and when changes are made to one nobody will remember the other and issues will pop up.

You can make it public if you want to, otherwise add an issue for it and it will be prioritized with all of the other enhancements.